src/tests/electron/window-frame-updater.test.ts

```diff
@@ -24,22 +24,70 @@ describe(WindowFrameUpdater, () => {
             });
 
         testSubject = new WindowFrameUpdater(windowStateStoreMock.object, browserWindowMock.object);
-        testSubject.initialize();
     });
 
-    it('sets window state to deviceConnectView', async () => {
-        browserWindowMock.setup(b => b.setSize(600, 391)).verifiable(Times.once());
+    describe('initialize', () => {
+        test.each(['deviceConnectView', 'resultsView'] as ViewRoutes[])('updates window based on initial state', async routeId => {
+            if (routeId === 'resultsView') {
```

Comment:
can this changing behaviour be pass through the "each" array? having an if-else inside the test doesn't seem like a good practice and it won't scale well when the time comes to add more views.